{\rtf1\ansi\ansicpg1252\cocoartf1038\cocoasubrtf360
{\fonttbl\f0\fswiss\fcharset0 Helvetica;}
{\colortbl;\red255\green255\blue255;}
\margl1440\margr1440\vieww9000\viewh8400\viewkind0
\pard\tx720\tx1440\tx2160\tx2880\tx3600\tx4320\tx5040\tx5760\tx6480\tx7200\tx7920\tx8640\ql\qnatural\pardirnatural

\f0\fs24 \cf0 diff -bru wxWidgets.orig/samples/minimal/minimal.cpp wxWidgets/samples/minimal/minimal.cpp\
--- wxWidgets.orig/samples/minimal/minimal.cpp	2014-07-31 21:16:45.000000000 -0700\
+++ wxWidgets/samples/minimal/minimal.cpp	2014-08-04 18:44:27.000000000 -0700\
@@ -67,6 +67,10 @@\
     // event handlers (these functions should _not_ be virtual)\
     void OnQuit(wxCommandEvent& event);\
     void OnAbout(wxCommandEvent& event);\
+    void OnMouseDown(wxMouseEvent &event);\
+    void OnMouseUp(wxMouseEvent &event);\
+    void OnRightDown(wxMouseEvent &event);\
+    void OnRightUp(wxMouseEvent &event);\
 \
 private:\
     // any class wishing to process wxWidgets events must use this macro\
@@ -99,6 +103,10 @@\
 wxBEGIN_EVENT_TABLE(MyFrame, wxFrame)\
     EVT_MENU(Minimal_Quit,  MyFrame::OnQuit)\
     EVT_MENU(Minimal_About, MyFrame::OnAbout)\
+    EVT_LEFT_DOWN(MyFrame::OnMouseDown)\
+    EVT_LEFT_UP(MyFrame::OnMouseUp)\
+    EVT_RIGHT_DOWN(MyFrame::OnRightDown)\
+    EVT_RIGHT_UP(MyFrame::OnRightUp)\
 wxEND_EVENT_TABLE()\
 \
 // Create a new application object: this macro will allow wxWidgets to create\
@@ -176,6 +184,61 @@\
 \
 \
 // event handlers\
+void MyFrame::OnMouseDown(wxMouseEvent &event)\
+\{\
+    wxString str;\
+    if( event.LeftIsDown() )\
+       str = "Left is down";\
+    else\
+       str = "Left is up";\
+    if( event.RightIsDown() )\
+       str += "Right is down";\
+    else\
+       str += "Right is up";\
+    GetStatusBar()->SetStatusText( str );\
+\}\
+\
+void MyFrame::OnMouseUp(wxMouseEvent &event)\
+\{\
+    wxString str;\
+    if( event.LeftIsDown() )\
+       str = "Left is down";\
+    else\
+       str = "Left is up";\
+    if( event.RightIsDown() )\
+       str += "Right is down";\
+    else\
+       str += "Right is up";\
+    GetStatusBar()->SetStatusText( str );\
+\}\
+\
+void MyFrame::OnRightDown(wxMouseEvent &event)\
+\{\
+    wxString str;\
+    if( event.LeftIsDown() )\
+       str = "Left is down";\
+    else\
+       str = "Left is up";\
+    if( event.RightIsDown() )\
+       str += "Right is down";\
+    else\
+       str += "Right is up";\
+    GetStatusBar()->SetStatusText( str );\
+\}\
+\
+void MyFrame::OnRightUp(wxMouseEvent &event)\
+\{\
+    wxString str;\
+    if( event.LeftIsDown() )\
+       str = "Left is down";\
+    else\
+       str = "Left is up";\
+    if( event.RightIsDown() )\
+       str += "Right is down";\
+    else\
+       str += "Right is up";\
+    GetStatusBar()->SetStatusText( str );\
+\}\
 \
 void MyFrame::OnQuit(wxCommandEvent& WXUNUSED(event))\
 \{\
}